tiledb/sm/rest/curl.cc

```diff
@@ -369,6 +369,8 @@ Status Curl::set_headers(struct curl_slist** headers) const {
     std::string basic_auth = username + std::string(":") + password;
     curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
     curl_easy_setopt(curl, CURLOPT_USERPWD, basic_auth.c_str());
+  } else {
+    curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_NONE);
```

Comment:
hmm. Why allow that? This doesn't look good or necessary. Why allow version checking to be unauthenticated?

Reply:
The endpoint doesn't require authentication server side, all other endpoints currently supported by core do. If we make this request we could assume we plan to make requests to authenticated routes so the error is likely inevitable, on first pass it just seemed strange to require a token for an unauthenticated route.

I removed it and we now require the authentication to be configured, I don't think there is really a case where we would just make this request and not send any others. I'll bring this up on my REST PR as well to see if we want to require authentication on the REST side as well.

Reply:
I'd go for requiring authentication on the REST side as well, but yes checking with REST experts might be a better idea. The only downside I can see is performance, but having a general rule that all client-server communication needs to be authenticated sounds more important to me.